tensorflow/tools/pip_package/utils/tf_wheel.bzl

```diff
@@ -23,17 +23,18 @@ Should be set via --repo_env=WHEEL_NAME=tensorflow_cpu.
 6) `--xla_aot` - paths to files that should be in xla_aot directory. 
 """
 
-load("@python_version_repo//:py_version.bzl", "WHEEL_COLLAB", "WHEEL_NAME")
+load("@python_version_repo//:py_version.bzl", "WHEEL_COLLAB", "WHEEL_NAME", "OUTPUT_PATH")
 load("//tensorflow:tensorflow.bzl", "VERSION")
 
 def _tf_wheel_impl(ctx):
     executable = ctx.executable.wheel_binary
 
     output = ctx.actions.declare_directory("wheel_house")
+    output_path = OUTPUT_PATH if OUTPUT_PATH else output.path
     args = ctx.actions.args()
     args.add("--project-name", WHEEL_NAME)
     args.add("--collab", str(WHEEL_COLLAB))
-    args.add("--output-name", output.path)
+    args.add("--output-name", output_path)
```

Comment:
To the best of my understanding this should make action on line 56 fail, because it declares that it outputs a directory (on line 59) which it would not actually create if output_path is chosen instead of output.path on line 33.

Also, this attempts to use a rule in a non-conventional way, exactly what we were trying to prevent by introducing this rule - the wheel must be created under bazel's cache, not under some random directory on file system.

Sorry, we can't let this PR in. If you need a wheel to be created in a different directory, you should add a simple post-build script which will simply copy it there. Please do not attempt to do it in the rule itself.

Probably it would be helpful to explain why passing WHEEL_NAME as an env var is ok-ish (tough I don't like it too, tbh), and passing OUTPUT_PATH is not ok. Wheel name selects only the name of resultant files, but regardless what the names they all of them will be placed under bazel's cache keeping things hermetic. In case of OUTPUT_PATH now it can put stuff outside of the cache in random location breaking hermeticity. Also, to the best of my understanding it should simply not work, because action on 56 should complain that the requested directory "wheel_house" was not actually created (if it doesn't then it is some bug in the original rule).

Reply:
Should hermeticity apply to the output artifacts as well? 
The main reason for introducing this was the wheel location within bazel cache was changing quite often and the post copying script needed to change accordingly.
Secondly, when using docker containers to build tf wheels, we cannot access the symblic link to bazel cache within container.

Reply:
> Should hermeticity apply to the output artifacts as well?
Yes, they should, in fact all bazel build artifacts, regardless if they are temporary or intermediate, are always output in cache. That is why i'm saying that to the best of my understanding this PR would not work, as it is trying to do what is explicitly forbidden by bazel core architecture decision: `bazel build` never writes anything outside of bazel cache. 

> the wheel location within bazel cache was changing quite often and the post copying script needed to change accordingly.

I'm surprised to hear this. It did change once recently, that was exactly when we introduced the `wheel` rule. Before that it was stable and it will remain stable from now on.

> Secondly, when using docker containers to build tf wheels, we cannot access the symblic link to bazel cache within container.

We are on the way of not needing to ever do that (using containers), as with proper hermetic build building it from container or building it from you raw system would make no difference,a s hermetic build does not care about local environment. The wheel creation is part of that. Whe have hermetic python introduced, we are on final stages of pushing hermetic cuda (https://github.com/openxla/xla/pull/10673). After that we will cleanup the devtoolset9 stuff from our images and likely bump manlinux tag for released pacakges. This will bring us in a situation when the only real local dependency you would need would be clang (and even that will possible become hermetic if we find time to do that).